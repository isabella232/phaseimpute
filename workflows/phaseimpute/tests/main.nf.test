nextflow_pipeline {

    name "Test phaseimpute workflow"
    script "main.nf"
    tag "pipeline"
    tag "pipeline/phaseimpute"


    test("Check test") {
        tag "test"
        config "../../../conf/test.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/imputation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("$outputDir/imputation/glimpse1/all.batch0.glimpse1.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("$outputDir/imputation/glimpse1/all.batch0.glimpse1.vcf.gz").vcf.header.getGenotypeSamples().sort()
                ).match()
                }
            )
        }

    }

    test("Check test_glimpse2") {
        tag "test_glimpse2"
        config "../../../conf/test_glimpse2.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/imputation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("$outputDir/imputation/glimpse2/all.batch0.glimpse2.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("$outputDir/imputation/glimpse2/all.batch0.glimpse2.vcf.gz").vcf.header.getGenotypeSamples().sort()
                ).match()
                }
            )
        }
    }

    test("Check test_quilt") {
        tag "test_quilt"
        config "../../../conf/test_quilt.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/imputation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("$outputDir/imputation/quilt/all.batch0.quilt.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("$outputDir/imputation/quilt/all.batch0.quilt.vcf.gz").vcf.header.getGenotypeSamples().sort()
                ).match()
                }
            )
        }

    }

    test("Check test_stitch") {
        tag "test_stitch"
        config "../../../conf/test_stitch.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/imputation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("$outputDir/imputation/stitch/all.batch0.stitch.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("$outputDir/imputation/stitch/all.batch0.stitch.vcf.gz").vcf.header.getGenotypeSamples().sort()
                ).match()
                }
            )
        }

    }

    test("Check test_sim") {
        tag "test_sim"
        config "../../../conf/test_sim.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/simulation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                ).match() }
            )
        }
    }

    test("Check test_validate") {
        tag "test_validate"
        config "../../../conf/test_validate.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/validation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten()
                ).match() }
            )
        }

    }

    test("Check test_all") {
        tag "test_all"
        config "../../../conf/test_all.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/simulation")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("${outputDir}/imputation")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("${outputDir}/prep_panel")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("${outputDir}/validation")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("${outputDir}/imputation/glimpse1/all.batch0.glimpse1.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/glimpse1/all.batch0.glimpse1.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/imputation/glimpse2/all.batch0.glimpse2.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/glimpse2/all.batch0.glimpse2.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/imputation/stitch/all.batch0.stitch.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/stitch/all.batch0.stitch.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/imputation/quilt/all.batch0.quilt.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/quilt/all.batch0.quilt.vcf.gz").vcf.header.getGenotypeSamples().sort()
                ).match()
                }
            )
        }
    }

    test("Check test_dog") {
        tag "test_dog"
        config "../../../conf/test_dog.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/prep_panel")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("${outputDir}/imputation")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                ).match()
                }
            )
        }
    }

    test("Check test_panelprep") {
        tag "test_panelprep"
        config "../../../conf/test_panelprep.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/prep_panel")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                ).match()
                }
            )
        }
    }

    test("Check test_batch") {
        tag "test_batch"
        config "../../../conf/test_batch.config"
        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/phaseimpute/'
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("${outputDir}/imputation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten(),
                    path("${outputDir}/imputation/glimpse2/all.batch0.glimpse2.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/glimpse2/all.batch0.glimpse2.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/imputation/quilt/all.batch0.quilt.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/quilt/all.batch0.quilt.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/imputation/glimpse2/all.batch1.glimpse2.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/glimpse2/all.batch1.glimpse2.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/imputation/quilt/all.batch1.quilt.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
                    path("${outputDir}/imputation/quilt/all.batch1.quilt.vcf.gz").vcf.header.getGenotypeSamples().sort(),
                    path("${outputDir}/validation/")
                        .list()
                        .collect { getRecursiveFileNames(it, outputDir) }
                        .flatten()
                ).match()
                }
            )
        }
    }
}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}
