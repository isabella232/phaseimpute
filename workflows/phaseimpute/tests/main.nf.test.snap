{
    "Check test_stitch": {
        "content": [
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/stats/NA12878.stitch.bcftools_stats.txt",
                "imputation/stats/NA19401.stitch.bcftools_stats.txt",
                "imputation/stats/NA20359.stitch.bcftools_stats.txt",
                "imputation/stitch/NA12878.batch0.stitch.vcf.gz",
                "imputation/stitch/NA12878.batch0.stitch.vcf.gz.tbi",
                "imputation/stitch/NA19401.batch0.stitch.vcf.gz",
                "imputation/stitch/NA19401.batch0.stitch.vcf.gz.tbi",
                "imputation/stitch/NA20359.batch0.stitch.vcf.gz",
                "imputation/stitch/NA20359.batch0.stitch.vcf.gz.tbi",
                "imputation/stitch/all.batch0.stitch.vcf.gz",
                "imputation/stitch/all.batch0.stitch.vcf.gz.tbi"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=false]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T21:57:43.929156494"
    },
    "Check test_all": {
        "content": [
            [
                "simulation/NA12878.depth_1x.bam",
                "simulation/NA12878.depth_1x.bam.csi",
                "simulation/NA19401.depth_1x.bam",
                "simulation/NA19401.depth_1x.bam.csi",
                "simulation/NA20359.depth_1x.bam",
                "simulation/NA20359.depth_1x.bam.csi",
                "simulation/csv/simulate.csv"
            ],
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/glimpse1/NA12878.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/NA12878.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse1/NA19401.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/NA19401.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse1/NA20359.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/NA20359.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse1/all.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/all.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse2/NA12878.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA12878.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/NA19401.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA19401.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/NA20359.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA20359.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz.tbi",
                "imputation/quilt/NA12878.batch0.quilt.vcf.gz",
                "imputation/quilt/NA12878.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/NA19401.batch0.quilt.vcf.gz",
                "imputation/quilt/NA19401.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/NA20359.batch0.quilt.vcf.gz",
                "imputation/quilt/NA20359.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/all.batch0.quilt.vcf.gz",
                "imputation/quilt/all.batch0.quilt.vcf.gz.tbi",
                "imputation/stats/NA12878.glimpse1.bcftools_stats.txt",
                "imputation/stats/NA12878.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA12878.quilt.bcftools_stats.txt",
                "imputation/stats/NA12878.stitch.bcftools_stats.txt",
                "imputation/stats/NA19401.glimpse1.bcftools_stats.txt",
                "imputation/stats/NA19401.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA19401.quilt.bcftools_stats.txt",
                "imputation/stats/NA19401.stitch.bcftools_stats.txt",
                "imputation/stats/NA20359.glimpse1.bcftools_stats.txt",
                "imputation/stats/NA20359.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA20359.quilt.bcftools_stats.txt",
                "imputation/stats/NA20359.stitch.bcftools_stats.txt",
                "imputation/stitch/NA12878.batch0.stitch.vcf.gz",
                "imputation/stitch/NA12878.batch0.stitch.vcf.gz.tbi",
                "imputation/stitch/NA19401.batch0.stitch.vcf.gz",
                "imputation/stitch/NA19401.batch0.stitch.vcf.gz.tbi",
                "imputation/stitch/NA20359.batch0.stitch.vcf.gz",
                "imputation/stitch/NA20359.batch0.stitch.vcf.gz.tbi",
                "imputation/stitch/all.batch0.stitch.vcf.gz",
                "imputation/stitch/all.batch0.stitch.vcf.gz.tbi"
            ],
            [
                "prep_panel/chunks/glimpse1/1000GP_chr21_chunks_glimpse1.txt",
                "prep_panel/chunks/glimpse1/1000GP_chr22_chunks_glimpse1.txt",
                "prep_panel/chunks/glimpse2/1000GP_chr21_chunks_glimpse2.txt",
                "prep_panel/chunks/glimpse2/1000GP_chr22_chunks_glimpse2.txt",
                "prep_panel/csv/chunks.csv",
                "prep_panel/csv/panel.csv",
                "prep_panel/csv/posfile.csv",
                "prep_panel/haplegend/1000GP_chr21.hap.gz",
                "prep_panel/haplegend/1000GP_chr21.legend.gz",
                "prep_panel/haplegend/1000GP_chr21.samples",
                "prep_panel/haplegend/1000GP_chr22.hap.gz",
                "prep_panel/haplegend/1000GP_chr22.legend.gz",
                "prep_panel/haplegend/1000GP_chr22.samples",
                "prep_panel/panel/1000GP_chr21_phased.vcf.gz",
                "prep_panel/panel/1000GP_chr21_phased.vcf.gz.csi",
                "prep_panel/panel/1000GP_chr22_phased.vcf.gz",
                "prep_panel/panel/1000GP_chr22_phased.vcf.gz.csi",
                "prep_panel/sites/1000GP_chr21_glimpse1_sites.vcf.gz",
                "prep_panel/sites/1000GP_chr21_glimpse1_sites.vcf.gz.tbi",
                "prep_panel/sites/1000GP_chr22_glimpse1_sites.vcf.gz",
                "prep_panel/sites/1000GP_chr22_glimpse1_sites.vcf.gz.tbi",
                "prep_panel/stats/1000GP.panel.bcftools_stats.txt"
            ],
            [
                "validation/samples/NA12878.batch0.truth.vcf.gz",
                "validation/samples/NA12878.batch0.truth.vcf.gz.tbi",
                "validation/samples/NA19401.batch0.truth.vcf.gz",
                "validation/samples/NA19401.batch0.truth.vcf.gz.tbi",
                "validation/samples/NA20359.batch0.truth.vcf.gz",
                "validation/samples/NA20359.batch0.truth.vcf.gz.tbi",
                "validation/samples/all.batch0.truth.vcf.gz",
                "validation/samples/all.batch0.truth.vcf.gz.tbi",
                "validation/stats/AllSamples.txt",
                "validation/stats/NA12878.truth.bcftools_stats.txt",
                "validation/stats/NA12878_P1000GP_Tglimpse1_SNP.txt",
                "validation/stats/NA12878_P1000GP_Tglimpse2_SNP.txt",
                "validation/stats/NA12878_P1000GP_Tquilt_SNP.txt",
                "validation/stats/NA12878_P1000GP_Tstitch_SNP.txt",
                "validation/stats/NA19401.truth.bcftools_stats.txt",
                "validation/stats/NA19401_P1000GP_Tglimpse1_SNP.txt",
                "validation/stats/NA19401_P1000GP_Tglimpse2_SNP.txt",
                "validation/stats/NA19401_P1000GP_Tquilt_SNP.txt",
                "validation/stats/NA19401_P1000GP_Tstitch_SNP.txt",
                "validation/stats/NA20359.truth.bcftools_stats.txt",
                "validation/stats/NA20359_P1000GP_Tglimpse1_SNP.txt",
                "validation/stats/NA20359_P1000GP_Tglimpse2_SNP.txt",
                "validation/stats/NA20359_P1000GP_Tquilt_SNP.txt",
                "validation/stats/NA20359_P1000GP_Tstitch_SNP.txt"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=false]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=true]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=false]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=true]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T22:05:34.729751585"
    },
    "Check test_validate": {
        "content": [
            [
                "validation/stats/AllSamples.txt",
                "validation/stats/NA12878.truth.bcftools_stats.txt",
                "validation/stats/NA12878_SNP.txt",
                "validation/stats/NA19401.truth.bcftools_stats.txt",
                "validation/stats/NA19401_SNP.txt",
                "validation/stats/NA20359.truth.bcftools_stats.txt",
                "validation/stats/NA20359_SNP.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T21:59:57.586965245"
    },
    "Check test_batch": {
        "content": [
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/batch/all.batch1.id.txt",
                "imputation/batch/all.batch1.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/glimpse2/NA12878.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA12878.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/NA19401.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA19401.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/all.batch1.glimpse2.vcf.gz",
                "imputation/glimpse2/all.batch1.glimpse2.vcf.gz.tbi",
                "imputation/quilt/NA12878.batch0.quilt.vcf.gz",
                "imputation/quilt/NA12878.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/NA19401.batch0.quilt.vcf.gz",
                "imputation/quilt/NA19401.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/all.batch0.quilt.vcf.gz",
                "imputation/quilt/all.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/all.batch1.quilt.vcf.gz",
                "imputation/quilt/all.batch1.quilt.vcf.gz.tbi",
                "imputation/stats/NA12878.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA12878.quilt.bcftools_stats.txt",
                "imputation/stats/NA19401.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA19401.quilt.bcftools_stats.txt",
                "imputation/stats/all.glimpse2.bcftools_stats.txt",
                "imputation/stats/all.quilt.bcftools_stats.txt"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=2, variantCount=1739, phased=true]",
            [
                "NA12878",
                "NA19401"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=2, variantCount=1739, phased=true]",
            [
                "NA12878",
                "NA19401"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=1, variantCount=1739, phased=true]",
            [
                "NA20359"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=1, variantCount=1739, phased=true]",
            [
                "NA20359"
            ],
            [
                "validation/stats/AllSamples.txt",
                "validation/stats/NA12878.truth.bcftools_stats.txt",
                "validation/stats/NA12878_P1000GP_Tglimpse2_SNP.txt",
                "validation/stats/NA12878_P1000GP_Tquilt_SNP.txt",
                "validation/stats/NA19401.truth.bcftools_stats.txt",
                "validation/stats/NA19401_P1000GP_Tglimpse2_SNP.txt",
                "validation/stats/NA19401_P1000GP_Tquilt_SNP.txt",
                "validation/stats/NA20359.truth.bcftools_stats.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T22:13:41.84665375"
    },
    "Check test_quilt": {
        "content": [
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/quilt/NA12878.batch0.quilt.vcf.gz",
                "imputation/quilt/NA12878.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/NA19401.batch0.quilt.vcf.gz",
                "imputation/quilt/NA19401.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/NA20359.batch0.quilt.vcf.gz",
                "imputation/quilt/NA20359.batch0.quilt.vcf.gz.tbi",
                "imputation/quilt/all.batch0.quilt.vcf.gz",
                "imputation/quilt/all.batch0.quilt.vcf.gz.tbi",
                "imputation/stats/NA12878.quilt.bcftools_stats.txt",
                "imputation/stats/NA19401.quilt.bcftools_stats.txt",
                "imputation/stats/NA20359.quilt.bcftools_stats.txt"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=true]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T21:56:22.270632603"
    },
    "Check test_sim": {
        "content": [
            [
                "simulation/NA12878.depth_1x.bam",
                "simulation/NA12878.depth_1x.bam.csi",
                "simulation/NA19401.depth_1x.bam",
                "simulation/NA19401.depth_1x.bam.csi",
                "simulation/NA20359.depth_1x.bam",
                "simulation/NA20359.depth_1x.bam.csi",
                "simulation/csv/simulate.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T21:58:40.027341891"
    },
    "Check test_panelprep": {
        "content": [
            [
                "prep_panel/chunks/glimpse1/1000GP_chr21_chunks_glimpse1.txt",
                "prep_panel/chunks/glimpse1/1000GP_chr22_chunks_glimpse1.txt",
                "prep_panel/chunks/glimpse2/1000GP_chr21_chunks_glimpse2.txt",
                "prep_panel/chunks/glimpse2/1000GP_chr22_chunks_glimpse2.txt",
                "prep_panel/csv/chunks.csv",
                "prep_panel/csv/panel.csv",
                "prep_panel/csv/posfile.csv",
                "prep_panel/haplegend/1000GP_chr21.hap.gz",
                "prep_panel/haplegend/1000GP_chr21.legend.gz",
                "prep_panel/haplegend/1000GP_chr21.samples",
                "prep_panel/haplegend/1000GP_chr22.hap.gz",
                "prep_panel/haplegend/1000GP_chr22.legend.gz",
                "prep_panel/haplegend/1000GP_chr22.samples",
                "prep_panel/panel/1000GP_chr21_phased.vcf.gz",
                "prep_panel/panel/1000GP_chr21_phased.vcf.gz.csi",
                "prep_panel/panel/1000GP_chr22_phased.vcf.gz",
                "prep_panel/panel/1000GP_chr22_phased.vcf.gz.csi",
                "prep_panel/sites/1000GP_chr21_glimpse1_sites.vcf.gz",
                "prep_panel/sites/1000GP_chr21_glimpse1_sites.vcf.gz.tbi",
                "prep_panel/sites/1000GP_chr22_glimpse1_sites.vcf.gz",
                "prep_panel/sites/1000GP_chr22_glimpse1_sites.vcf.gz.tbi"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-14T19:19:50.07758554"
    },
    "Check test_glimpse2": {
        "content": [
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/glimpse2/NA12878.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA12878.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/NA19401.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA19401.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/NA20359.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/NA20359.batch0.glimpse2.vcf.gz.tbi",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz.tbi",
                "imputation/stats/NA12878.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA19401.glimpse2.bcftools_stats.txt",
                "imputation/stats/NA20359.glimpse2.bcftools_stats.txt"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=true]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T21:54:12.807711398"
    },
    "Check test": {
        "content": [
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/glimpse1/NA12878.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/NA12878.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse1/NA19401.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/NA19401.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse1/NA20359.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/NA20359.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse1/all.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/all.batch0.glimpse1.vcf.gz.tbi",
                "imputation/stats/NA12878.glimpse1.bcftools_stats.txt",
                "imputation/stats/NA19401.glimpse1.bcftools_stats.txt",
                "imputation/stats/NA20359.glimpse1.bcftools_stats.txt"
            ],
            "VcfFile [chromosomes=[chr21, chr22], sampleCount=3, variantCount=1739, phased=false]",
            [
                "NA12878",
                "NA19401",
                "NA20359"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T21:52:54.602243735"
    },
    "Check test_dog": {
        "content": [
            [
                "prep_panel/chunks/glimpse1/658Dog_chr21_chunks_glimpse1.txt",
                "prep_panel/chunks/glimpse1/658Dog_chr22_chunks_glimpse1.txt",
                "prep_panel/chunks/glimpse2/658Dog_chr21_chunks_glimpse2.txt",
                "prep_panel/chunks/glimpse2/658Dog_chr22_chunks_glimpse2.txt",
                "prep_panel/csv/chunks.csv",
                "prep_panel/csv/panel.csv",
                "prep_panel/csv/posfile.csv",
                "prep_panel/haplegend/658Dog_chr21.hap.gz",
                "prep_panel/haplegend/658Dog_chr21.legend.gz",
                "prep_panel/haplegend/658Dog_chr21.samples",
                "prep_panel/haplegend/658Dog_chr22.hap.gz",
                "prep_panel/haplegend/658Dog_chr22.legend.gz",
                "prep_panel/haplegend/658Dog_chr22.samples",
                "prep_panel/panel/658Dog_chr21_phased.vcf.gz",
                "prep_panel/panel/658Dog_chr21_phased.vcf.gz.csi",
                "prep_panel/panel/658Dog_chr22_phased.vcf.gz",
                "prep_panel/panel/658Dog_chr22_phased.vcf.gz.csi",
                "prep_panel/sites/658Dog_chr21_glimpse1_sites.vcf.gz",
                "prep_panel/sites/658Dog_chr21_glimpse1_sites.vcf.gz.tbi",
                "prep_panel/sites/658Dog_chr22_glimpse1_sites.vcf.gz",
                "prep_panel/sites/658Dog_chr22_glimpse1_sites.vcf.gz.tbi"
            ],
            [
                "imputation/batch/all.batch0.id.txt",
                "imputation/batch/all.batch0.noid.txt",
                "imputation/csv/impute.csv",
                "imputation/glimpse1/dog_1735.batch0.glimpse1.vcf.gz",
                "imputation/glimpse1/dog_1735.batch0.glimpse1.vcf.gz.tbi",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz",
                "imputation/glimpse2/all.batch0.glimpse2.vcf.gz.tbi",
                "imputation/quilt/all.batch0.quilt.vcf.gz",
                "imputation/quilt/all.batch0.quilt.vcf.gz.tbi",
                "imputation/stats/all.glimpse2.bcftools_stats.txt",
                "imputation/stats/all.quilt.bcftools_stats.txt",
                "imputation/stats/dog_1735.glimpse1.bcftools_stats.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-12T22:07:58.047722839"
    }
}